vertical-pod-autoscaler/pkg/admission-controller/resource/vpa/matcher.go

```diff
@@ -52,9 +54,17 @@ func (m *matcher) GetMatchingVPA(pod *core.Pod) *vpa_types.VerticalPodAutoscaler
 	}
 	onConfigs := make([]*vpa_api_util.VpaWithSelector, 0)
 	for _, vpaConfig := range configs {
-		if vpa_api_util.GetUpdateMode(vpaConfig) == vpa_types.UpdateModeOff {
+		updateMode := vpa_api_util.GetUpdateMode(vpaConfig)
+		if updateMode == vpa_types.UpdateModeOff {
```

Comment:
What about other modes like:
UpdateModeInitial
UpdateModeRecreate
UpdateModeAuto

why don't you also continue in these ones ?

Reply:
I'm not changing the code here. This code finds "any vpa that doesn't look disabled"